### PR TITLE
internal/olm: get installed version from cluster instead of required CLI argument

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** get the version of OLM installed in a cluster without passing `--version=<version>` to `olm status`.

**Motivation for the change:** a user may not know which OLM version was installed in a cluster.

@joelanford we might be able to use `Client.GetInstalledVersion()` in `Uninstall()` too.

### CODE REVIEW

**Comment by joelanford** [comment]
@estroz Makes sense to me to use it for uninstall.

File: hack/tests/alpha-olm-subcommands.sh [hunk 1]
```diff
@@ -23,18 +23,12 @@ test_version() {
     echo $commandoutput | grep -F "Failed to install OLM version \\\"${version}\\\": detected existing OLM resources: OLM must be completely uninstalled before installation"
 
     # Status should succeed with OLM installed
-    # If version is "latest", also run without --version flag
-    if [[ "$version" == "latest" ]]; then
-        commandoutput=$(operator-sdk alpha olm status 2>&1)
-        echo $commandoutput | grep -F "Successfully got OLM status for version \\\"${version}\\\""
```
Comment (joelanford) [inline hack/tests/alpha-olm-subcommands.sh]: The primary purpose of this test is to make sure that `"latest"` is used as the default version when `--version` is not specified. Maybe we should move this logic to the `install` subcommand, and use `operator-sdk alpha olm install` when `$version == "latest"`?

**Comment by estroz** [comment]
@joelanford indeed they do: https://github.com/operator-framework/operator-lifecycle-manager/issues/935.

File: internal/olm/client.go [hunk 2]
```diff
@@ -164,6 +168,53 @@ func (c Client) UninstallVersion(ctx context.Context, version string) error {
...
+	return getOLMVersionFromPackageServerCSV(pkgServerCSV), nil
+}
+
+const (
+	packageServerCSVNamePrefix   = "packageserver."
+	packageServerOLMVersionLabel = "olm.version"
+)
+
+func getOLMVersionFromPackageServerCSV(csv *olmapiv1alpha1.ClusterServiceVersion) string {
+	// Package server CSV's from OLM versions > 0.10.1 have a label containing
```
Comment (estroz) [inline internal/olm/client.go]: @joelanford the way OLM stores version information [changed](https://github.com/operator-framework/operator-lifecycle-manager/issues/935#issuecomment-513856066) in OLM version `> 0.10.1`. Trying `labels['olm.version']` first, then falling back to package server CSV name if that doesn't exist (for OLM versions `<= 0.10.1`), should work.

File: internal/olm/manager.go [hunk 3]
```diff
@@ -108,18 +113,22 @@ func (m *Manager) Status() error {
...
+		return err
+	}
+
+	status, err := m.Client.GetStatus(ctx, version)
 	if err != nil {
 		return err
 	}
 
-	log.Infof("Successfully got OLM status for version %q", m.Version)
+	log.Infof("Successfully got OLM status for version %s", version)
```
Comment (joelanford) [inline internal/olm/manager.go]: Nit: any reason to change this to `%s` but leave the other "Successfully installed/uninstalled..." logs as `%q`?

**Comment by joelanford** [comment]
@estroz Can you also update `doc/sdk-cli-reference.md` and the CHANGELOG to reflect these changes?

File: internal/olm/manager.go (hunk 3 above)
Comment (estroz) [reply internal/olm/manager.go]: Nope, reverted.

**Pull request merged**